src/ansi-c/library/stdlib.c

```diff
@@ -115,6 +115,9 @@ inline void *malloc(__CPROVER_size_t malloc_size)
   // realistically, malloc may return NULL,
   // and __CPROVER_allocate doesn't, but no one cares
   __CPROVER_HIDE:;
+  // 8 is the default object bits for ANSI-C and C++
+  if (malloc_size >= ((__CPROVER_size_t)1<<(64-8)))
```

Comment:
Can't the object bits size be changed on the command line with `--object-bits` ? and if so, this probably needs to accommodate that. Ditto for the hardcoded `64`, which would probably be wrong if `--32` is specified to cbmc :-)

Reply:
I moved the check into `goto-checker`, i.e. the `stdlib.c` implemetation is unchanged, we can use the pointer-width and object-bits from config, and the whole thing is behind an option, so it won't affect the existing tests.